lib/ansible/module_utils/common/process.py

```diff
@@ -4,11 +4,36 @@
 from __future__ import annotations
 
 import os
+import threading
+
+from multiprocessing.pool import ThreadPool, Pool
+from multiprocessing.context import BaseContext
 
 from ansible.module_utils.common.file import is_executable
 from ansible.module_utils.common.warnings import deprecate
 
 
+class DummyContext(BaseContext):
+    def Lock(self):
+        return threading.Lock()
+
+
+class DummyPool(ThreadPool):
+    def __init__(self, processes=None, initializer=None, initargs=()):
+        Pool.__init__(self, processes, initializer, initargs, context=DummyContext())
+
+
+def get_thread_pool(size, warn=None):
```

Comment:
Until we have a use case for making this utility code, we should probably keep it within the fact gathering code that needs it.

Reply:
if it were the single function, I would agree, but if we add this kind of required scafolding, it makes little sense to keep there.

Reply:
If we end up needing it in other modules later, we can move it into module_utils. Even then, unless there's a need to make it part of our public API, it should be private.